### PR TITLE
Add code_signing_allowed config property

### LINKED ISSUE #30: No option to toggle CODE_SIGNING_ALLOWED
### Troubleshooting

- [x] I've searched [discuss.bitrise.io](https://discuss.bitrise.io/) for possible solutions.
- Which version of the step is affected? 0.11.6
- Is the issue reproducible with the latest version? YES
- Does the issue happen sporadically, or every time? EVERY TIME
- Is the issue reproducible locally by following our [local debug guide](https://devcenter.bitrise.io/troubleshooting/debugging-your-build-on-your-own-machine/)? YES

### Useful information

- [FAQ](https://devcenter.bitrise.io/faq/faq-index/)
- [devcenter.bitrise.io](https://devcenter.bitrise.io/)

### Issue description

Currently, there is no option to toggle "CODE_SIGNING_ALLOWED=NO", which is required by some CocoaPod frameworks such as Firebase Remote Config to work properly.

### Bitrise info

- Build URL: [_Please share the build URL with us._](https://app.bitrise.io/build/348083c9-69b9-42a1-8d20-dd94bf003465)
- [Bitrise Support](https://devcenter.bitrise.io/troubleshooting/enabling-bitrise-support-user/) enabled: NO
- Log: Xcode build for simulator

<details>
  <pre>
Running build
$ set -o pipefail && xcodebuild "-workspace" "/Users/vagrant/git/MyProject.xcworkspace" "-scheme" "My Project" "-configuration" "QA" "CODE_SIGNING_ALLOWED=NO" "-destination" "id=5210DB4C-7CD4-4DE7-AC08-6E2F259F04B4" "COMPILER_INDEX_STORE_ENABLE=NO" "clean" "build" | xcpretty
  </pre>
</details>
It is possible to work around this limitation by adding "CODE_SIGNING_ALLOWED=YES" in the 'Additional options for xcodebuild call' under the 'Debug' section but I wouldn't rely on this as it produces a command that looks like this with two CODE_SIGNING_ALLOWED flags that may not always work:

```
xcodebuild "-workspace" "/Users/vagrant/git/MyProject.xcworkspace" "-scheme" "My Project" "-configuration" "QA" "CODE_SIGNING_ALLOWED=NO" "-destination" "id=5210DB4C-7CD4-4DE7-AC08-6E2F259F04B4" "COMPILER_INDEX_STORE_ENABLE=NO" "clean" "build" "CODE_SIGNING_ALLOWED=YES" | xcpretty
```

### Steps to reproduce

Run the build for the simulator step on a project that uses Firebase Remote Config key values. The values should be fetched on app launch but are never called.

### PR DESCRIPTION
**Pull request by hisaac**
Closes https://github.com/bitrise-steplib/steps-xcode-build-for-simulator/issues/30

### Checklist

- [x] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a MINOR update

### Context

Some builds need to allow codesigning when building for the simulator, but the step in its current state does not allow this to be configured easily.

### Changes

- Adds a `codes_signing_allowed` configuration option
- Passes that option to the `SetDisableCodesign` method
- fixes #30 issue

### CODE REVIEW

**Comment by hisaac** [comment]
@shams-ahmed Still more work to do I'm sure, but wanted to get this up to make sure I'm headed in the right direction.

File: step.yml [hunk 1]
```diff
@@ -157,6 +157,13 @@ inputs:
       value_options:
         - "yes"
         - "no"
+  - code_signing_allowed: "no"
+    opts:
+      title: Code Signing Allowed
+      description: Whether codesigning is allowed for this build.
```
Comment (shams-ahmed) [inline step.yml:163]: can you extend the description to state why it should be set to NO for the simulator

Comment (hisaac) [reply step.yml:163]: I don't actually know the answer to that 😬 What is the reason why we have it codesigning disabled by default?

Comment (shams-ahmed) [reply step.yml:163]:
```
title: Enable code signing
description: When building an app for the simulator, code signing is not required and is set to "no" by default.
Only on rare occasions do you need to set the flag to "yes", usually when working with certain test cases or third-party dependency
```

Comment (hisaac) [reply step.yml]: Thanks for the help on the description. Updated in https://github.com/bitrise-steplib/steps-xcode-build-for-simulator/pull/31/commits/bcb9fcb2c31fb1e566fa36fc9b97619bc477099d

**Review by shams-ahmed — DISMISSED** [review]

**Review by shams-ahmed — APPROVED** [review]

**Pull request merged**